src/test/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseServiceTest.java

```diff
@@ -115,6 +120,31 @@ public void saveResponseRejectionWithoutDirectionsQuestionnaire() {
         verify(caseRepository, never()).saveCaseEvent(AUTHORISATION, claim, REFERRED_TO_MEDIATION);
     }
 
+    @Test
+    public void saveCaseEventCaseStayedAfterFullDefenseDispute() {
+
+        ClaimantResponse claimantResponse = SampleClaimantResponse
+            .ClaimantResponseAcceptation
+            .builder()
+            .build();
+
+        Claim claim = SampleClaim.builder()
+            .withResponse(SampleResponse.FullDefence.builder().withDefenceType(DefenceType.DISPUTE).build())
+            .withRespondedAt(LocalDateTime.now())
+            .withClaimantResponse(claimantResponse)
+            .build();
+
+        when(claimService.getClaimByExternalId(eq(EXTERNAL_ID), eq(AUTHORISATION))).thenReturn(claim);
+        when(caseRepository.saveClaimantResponse(any(Claim.class), any(ResponseAcceptation.class), eq(AUTHORISATION)))
+            .thenReturn(claim);
+
+        claimantResponseService.save(EXTERNAL_ID, claim.getSubmitterId(), claimantResponse, AUTHORISATION);
+
+        verify(appInsights, once()).trackEvent(eq(CLAIM_STAYED), eq(REFERENCE_NUMBER), eq(claim.getReferenceNumber()));
+        verify(caseRepository).saveCaseEvent(eq(AUTHORISATION), eq(claim), eq(STAY_CLAIM));
```

Comment:
Just a reminder: when every parameter is matched with `eq()`, then they can all be just the original values
`.trackEvent(STAYED_CLAIM, REFERENCE_NUMBER, claim.getReferenceNumber())`